src/main/java/minicraft/item/Inventory.java

```diff
@@ -106,32 +107,47 @@ public int add(int slot, Item item) {
 						adding.count = i + 1 == c && toTake.count % 100 > 0 ? toTake.count % 100 : 100;
 						if (adding.count == 0) break;
 						if (items.size() == maxItem) return total - toTake.count;
-						items.add(adding); // Add the item to the items list
+						addItemToList(adding); // Add the item to the items list
 						toTake.count -= adding.count;
 					}
 					return total;
 				} else {
 					return total - toTake.count;
 				}
 			} else {
-				items.add(slot, toTake);
+				addItemToList(slot, toTake);
 				return total;
 			}
 		}
 
 		if (!unlimited) {
 			if (items.size() < maxItem) {
-				items.add(slot, item); // Add the item to the items list
+				addItemToList(slot, item); // Add the item to the items list
 				return 1;
 			} else {
 				return 0;
 			}
 		} else {
-			items.add(slot, item);
+			addItemToList(slot, item);
 			return 1;
 		}
 	}
 
+	private void addItemToList(Item item) { addItemToList(items.size(), item); }
+	private void addItemToList(int index, Item item) {
```

Comment:
These functions should be renamed so they explain _exactly_ what they are doing. This name does not tell us that it checks for quests.

Good that it is private.